xlators/debug/io-stats/src/io-stats.c

```diff
@@ -3757,6 +3757,14 @@ reconfigure(xlator_t *this, dict_t *options)
     ios_set_log_format_code(conf, dump_format_str);
     GF_OPTION_RECONF("ios-sample-buf-size", conf->ios_sample_buf_size, options,
                      int32, out);
+    if (!conf->ios_sample_interval &&
+        (conf->ios_sample_buf_size == GF_IOS_SAMPLE_BUF_SIZE)) {
+        conf->ios_sample_buf_size = 1024;
```

Comment:
This makes impossible to configure a buffer size of GF_IOS_SAMPLE_BUF_SIZE. We should detect if we are using the default value in another way.

Reply:
It is configure a buffer size of GF_IOS_SAMPLE_BUF_SIZE while ios_sample_interval is configured otherwise not.

Reply:
But when `ios_sample_interval` is configured, we allow `GF_IOS_SAMPLE_BUF_SIZE - 1` and `GF_IOS_SAMPLE_BUF_SIZE + 1`. Then why `GF_IOS_SAMPLE_BUF_SIZE` is not allowed ? either we allow all values or we don't allow any when `ios_sample_interval` is configured.